crates/nu-cli/src/commands/run_external.rs

```diff
@@ -120,3 +170,54 @@ impl WholeStreamCommand for RunExternalCommand {
         Ok(stream.to_output_stream())
     }
 }
+
+#[allow(unused_variables)]
+async fn maybe_autocd_dir<'a>(cmd: &ExternalCommand, ctx: &mut Context) -> Option<String> {
+    // We will "auto cd" if
+    //   - the command name ends in a path separator, or
+    //   - it's not a command on the path and no arguments were given.
+    let name = &cmd.name;
+    let path_name = if name.ends_with(std::path::MAIN_SEPARATOR)
+        || (cmd.args.is_empty()
+            && PathBuf::from(name).is_dir()
+            && dunce::canonicalize(name).is_ok()
+            && ichwh::which(&name).await.unwrap_or(None).is_none())
```

Comment:
I tried to break up the `||` conditions with an `else if`, but clippy doesn't like two branches having the same body.